applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -323,57 +323,64 @@ static inline int adjust_rsrq(int input)
 	return input;
 }
 
-static void send_neighbor_cell_update(struct lte_lc_cells_info *cell_info)
+static void send_cloud_location_update(const struct location_data_cloud *cloud_location_info)
 {
 	struct location_module_event *evt = new_location_module_event();
 
-	BUILD_ASSERT(sizeof(evt->data.neighbor_cells.cell_data) ==
-		     sizeof(struct lte_lc_cells_info));
-	BUILD_ASSERT(sizeof(evt->data.neighbor_cells.neighbor_cells) >=
-		     sizeof(struct lte_lc_ncell) * CONFIG_LTE_NEIGHBOR_CELLS_MAX);
-
-	memcpy(&evt->data.neighbor_cells.cell_data, cell_info, sizeof(struct lte_lc_cells_info));
-	memcpy(&evt->data.neighbor_cells.neighbor_cells, cell_info->neighbor_cells,
-	       sizeof(struct lte_lc_ncell) * evt->data.neighbor_cells.cell_data.ncells_count);
-
-	/* Convert RSRP to dBm and RSRQ to dB per "nRF91 AT Commands" v1.7. */
-	evt->data.neighbor_cells.cell_data.current_cell.rsrp =
-			adjust_rsrp(evt->data.neighbor_cells.cell_data.current_cell.rsrp);
-	evt->data.neighbor_cells.cell_data.current_cell.rsrq =
-			adjust_rsrq(evt->data.neighbor_cells.cell_data.current_cell.rsrq);
-
-	for (size_t i = 0; i < evt->data.neighbor_cells.cell_data.ncells_count; i++) {
-		evt->data.neighbor_cells.neighbor_cells[i].rsrp =
-			adjust_rsrp(evt->data.neighbor_cells.neighbor_cells[i].rsrp);
-		evt->data.neighbor_cells.neighbor_cells[i].rsrq =
-			adjust_rsrq(evt->data.neighbor_cells.neighbor_cells[i].rsrq);
+	if (cloud_location_info->cell_data != NULL) {
+		evt->data.cloud_location.neighbor_cells_valid = true;
+		BUILD_ASSERT(sizeof(evt->data.cloud_location.neighbor_cells.cell_data) ==
+			     sizeof(struct lte_lc_cells_info));
+		BUILD_ASSERT(sizeof(evt->data.cloud_location.neighbor_cells.neighbor_cells) >=
+			     sizeof(struct lte_lc_ncell) * CONFIG_LTE_NEIGHBOR_CELLS_MAX);
+
+		memcpy(&evt->data.cloud_location.neighbor_cells.cell_data,
+		       cloud_location_info->cell_data, sizeof(struct lte_lc_cells_info));
+		memcpy(&evt->data.cloud_location.neighbor_cells.neighbor_cells,
+		       cloud_location_info->cell_data->neighbor_cells,
+		       sizeof(struct lte_lc_ncell) *
+				evt->data.cloud_location.neighbor_cells.cell_data.ncells_count);
+
+		/* Convert RSRP to dBm and RSRQ to dB per "nRF91 AT Commands" v1.7. */
+		evt->data.cloud_location.neighbor_cells.cell_data.current_cell.rsrp =
+			adjust_rsrp(evt->data.cloud_location.neighbor_cells
+				.cell_data.current_cell.rsrp);
+		evt->data.cloud_location.neighbor_cells.cell_data.current_cell.rsrq =
+			adjust_rsrq(evt->data.cloud_location.neighbor_cells
+				.cell_data.current_cell.rsrq);
+
+		for (size_t i = 0;
+		     i < evt->data.cloud_location.neighbor_cells.cell_data.ncells_count;
```

Comment:
This one is a little bit ackward with the prolonged naming. Since its only a count you could write this value to a uin8_t with shorter naming and iterate over that variable,

Reply:
I've added a shortcut into this function so I can clean the entire function a bit:
`struct location_module_neighbor_cells *ncells = &evt->data.cloud_location.neighbor_cells;`